### PR TITLE
Add org.wfview.wfview

### PR DESCRIPTION
**Pull request by AsciiWolf**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://gitlab.com/eliggett/wfview/-/issues/138#note_1650734121
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
https://gitlab.com/eliggett/wfview/-/merge_requests/19

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

File: org.wfview.wfview.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+app-id: org.wfview.wfview
+runtime: org.kde.Platform
+runtime-version: 5.15-23.08
```
Comment (hfiguiere) [inline org.wfview.wfview.yaml:3]: why not using the 6.6 runtime?

Comment (AsciiWolf) [reply org.wfview.wfview.yaml:3]: Good question. I will try updating it. :-)

Comment (eliggett) [reply org.wfview.wfview.yaml:3]: We've had wfview building with qt6 on windows for some time, it should be alright with Linux as well.

File: org.wfview.wfview.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+sdk: org.kde.Sdk
+command: wfview
+rename-desktop-file: wfview.desktop
+finish-args:
+  - --device=all
```
Comment (AsciiWolf) [inline org.wfview.wfview.yaml:8]: Required for serial devices etc.

File: qcustomplot-add-cmake.patch [hunk 3]
```diff
@@ -0,0 +1,145 @@
+--- /dev/null
```
Comment (AsciiWolf) [inline qcustomplot-add-cmake.patch:1]: QCustomPlot is distributed without any build system manifest (there is a sample qmake file in additional "examples" archive). This is intentional and QCustomPlot upstream is not interested in the CMake file.

**Comment by AsciiWolf** [comment]
bot, build org.wfview.wfview

File: org.wfview.wfview.yaml (hunk 1 above)
Comment (AsciiWolf) [reply org.wfview.wfview.yaml]: Done. Let's see if it builds. :-)

**Comment by AsciiWolf** [comment]
bot, build org.wfview.wfview

**Comment by AsciiWolf** [comment]
bot, build org.wfview.wfview

File: org.wfview.wfview.yaml (hunk 1 above)
Comment (AsciiWolf) [reply org.wfview.wfview.yaml]: It sadly does not build with the 6.6 runtime:


```
wfmain.cpp:2617:40: error: conversion from ‘long int’ to ‘const QVariant’ is ambiguous
 2617 |             if (settings->value("Icon",NULL) != NULL) {
      |                                        ^~~~

wfmain.cpp:2618:63: error: conversion from ‘long int’ to ‘const QVariant’ is ambiguous
 2618 |                 butt.icon = new QImage(settings->value("Icon",NULL).value<QImage>());
      |                                                               ^~~~

...

wfmain.cpp:2618:81: error: expected primary-expression before ‘>’ token
 2618 |                 butt.icon = new QImage(settings->value("Icon",NULL).value<QImage>());
      |                                                                                 ^
wfmain.cpp:2618:83: error: expected primary-expression before ‘)’ token
 2618 |                 butt.icon = new QImage(settings->value("Icon",NULL).value<QImage>());
      |                                                                                   ^
```

@eliggett Any idea?

**Comment by eliggett** [comment]
Those lines that are failing to compile, the QImage stuff, comment it out. This is something we have to work on for qt6 I believe. Commenting them out will just remove the custom icons in the streamdeck controller.

**Comment by hfiguiere** [comment]
or just stick to 5.15 until ready.

**Comment by eliggett** [comment]
> or just stick to 5.15 until ready.

Could. It is compiling on my build farm. My linux build farm is still qt5. Windows I have qt 6.3. I'd say that, at least on linux, we have more testing on qt5 by far.

**Comment by eliggett** [comment]
For linux you will want qt 5.15. My specific version I use for testing is 5.15.3. Version 5.12 will give you issues.

**Comment by AsciiWolf** [comment]
Thanks for the info! I will stick with the 5.15 runtime for now.

File: org.wfview.wfview.png
Comment (AsciiWolf) [inline org.wfview.wfview.png:1]: Upstream does not provide a standard size desktop icon. This icon is resized from the more wide upstream one.

**Comment by AsciiWolf** [comment]
bot, build org.wfview.wfview

**Comment by xylo04** [comment]
@AsciiWolf I've used this before; I'll volunteer to co-maintain!

File: org.wfview.wfview.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.wfview.wfview</id>
+  <launchable type="desktop-id">org.wfview.wfview.desktop</launchable>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-or-later</project_license>
```
Comment (bbhtt) [inline org.wfview.wfview.metainfo.xml:6]: Add `developer_name`

File: org.wfview.wfview.metainfo.xml
Comment (bbhtt) [inline org.wfview.wfview.metainfo.xml:1]: This file should be upstream, can you submit it?

File: org.wfview.wfview.png
Comment (bbhtt) [reply org.wfview.wfview.png:1]: Probably should open an issue to provide one

Comment (bbhtt) [reply org.wfview.wfview.png:1]: Probably should open an issue to provide one

File: org.wfview.wfview.metainfo.xml
Comment (AsciiWolf) [reply org.wfview.wfview.metainfo.xml:1]: Done.

https://gitlab.com/eliggett/wfview/-/merge_requests/19

File: org.wfview.wfview.png
Comment (AsciiWolf) [reply org.wfview.wfview.png:1]: I did submit the 256x256 standard icon into upstream as a part of this MR:

https://gitlab.com/eliggett/wfview/-/merge_requests/19

**Comment by AsciiWolf** [comment]
The Flatpak should now be ready to be published. :-) I have [submitted](https://gitlab.com/eliggett/wfview/-/merge_requests/19) all downstream (desktop-related) changes into upstream.

**Pull request closed without merging**